OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -717,9 +717,10 @@ public bool InterpretCommand(S server, Connection conn, Session.Client client, s
 				{ "name",
 					s =>
 					{
-						Log.Write("server", "Player@{0} is now known as {1}.", conn.Socket.RemoteEndPoint, s);
-						server.SendMessage("{0} is now known as {1}.".F(client.Name, s));
-						client.Name = s;
+						var sanitizedName = OpenRA.Settings.SanitizedPlayerName(s);
+						Log.Write("server", "Player@{0} is now known as {1}.", conn.Socket.RemoteEndPoint, sanitizedName);
```

Comment:
Could you add a check if the name changed really?
If I try to change `abcdefg30` to `abcdefg30#` the `#` is removed and `abcdefg30 is now known as abcdefg30` appears.